exchanges/okx/okx_type_convert.go

```diff
@@ -0,0 +1,2495 @@
+		a.TakeProfitOrderPrice, err = strconv.ParseFloat(chil.TakeProfitOrderPrice, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.StopLossTriggerPrice != "" {
+		a.StopLossTriggerPrice, err = strconv.ParseFloat(chil.StopLossTriggerPrice, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.RebateAmount != "" {
+		a.RebateAmount, err = strconv.ParseFloat(chil.RebateAmount, 64)
+		if err != nil {
+			return err
+		}
+	}
+	switch strings.ToUpper(chil.InstrumentType) {
+	case okxInstTypeSwap:
+		a.InstrumentType = asset.PerpetualSwap
+	case okxInstTypeSpot:
+		a.InstrumentType = asset.Spot
+	case okxInstTypeFutures:
+		a.InstrumentType = asset.Futures
+	case okxInstTypeOption:
+		a.InstrumentType = asset.Option
+	case okxInstTypeContract:
+		a.InstrumentType = asset.PerpetualSwap
+	case okxInstTypeMargin:
+		a.InstrumentType = asset.Margin
+	case okxInstTypeANY:
+		a.InstrumentType = asset.Empty
+	}
+	return nil
+}
+
+// UnmarshalJSON decerializes JSON, and timestamp information.
+func (a *PendingOrderItem) UnmarshalJSON(data []byte) error {
+	type Alias PendingOrderItem
+	chil := &struct {
+		*Alias
+		Side                string `json:"side"`
+		UpdateTime          string `json:"uTime"`
+		CreationTime        string `json:"cTime"`
+		InstrumentType      string `json:"instType"`
+		AccumulatedFillSize string `json:"accFillSz"`
+		AveragePrice        string `json:"avgPx"`
+		LastFilledSize      string `json:"fillSz"`
+		Leverage            string `json:"lever"`
+	}{
+		Alias: (*Alias)(a),
+	}
+	err := json.Unmarshal(data, chil)
+	if err != nil {
+		return err
+	}
+	uTime, err := strconv.ParseInt(chil.UpdateTime, 10, 64)
+	if err != nil {
+		return err
+	}
+	cTime, err := strconv.ParseInt(chil.CreationTime, 10, 64)
+	if err != nil {
+		return err
+	}
+	a.Side, err = order.StringToOrderSide(chil.Side)
+	if err != nil {
+		return err
+	}
+	chil.InstrumentType = strings.ToUpper(chil.InstrumentType)
+	if chil.AccumulatedFillSize != "" {
+		a.AccumulatedFillSize, err = strconv.ParseFloat(chil.AccumulatedFillSize, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.AveragePrice != "" {
+		a.AveragePrice, err = strconv.ParseFloat(chil.AveragePrice, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.LastFilledSize != "" {
+		a.LastFilledSize, err = strconv.ParseFloat(chil.LastFilledSize, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.Leverage != "" {
+		a.Leverage, err = strconv.ParseFloat(chil.Leverage, 64)
+		if err != nil {
+			return err
+		}
+	}
+	switch strings.ToUpper(chil.InstrumentType) {
+	case okxInstTypeSwap:
+		a.InstrumentType = asset.PerpetualSwap
+	case okxInstTypeSpot:
+		a.InstrumentType = asset.Spot
+	case okxInstTypeFutures:
+		a.InstrumentType = asset.Futures
+	case okxInstTypeOption:
+		a.InstrumentType = asset.Option
+	case okxInstTypeContract:
+		a.InstrumentType = asset.PerpetualSwap
+	case okxInstTypeMargin:
+		a.InstrumentType = asset.Margin
+	case okxInstTypeANY:
+		a.InstrumentType = asset.Empty
+	}
```

Comment:
This can be a function to reduce code across this file. Please change all reference.